[Feature] Support dynamic modification of datacache.partition_duration

stonechen111 opened the pull request:
Modify the partition_duration parameter from read-only to dynamic modification. 
This facilitates users to modify table attributes in real time.

Fixes #32840

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/StarRocks/starrocks?pullRequest=32844) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks?pullRequest=32844) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/StarRocks/starrocks?pullRequest=32844) it.</sub>

wanpengfei-git commented:
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

wyb commented on fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:
```diff
@@ -1600,28 +1601,41 @@ public AlterJobV2 createAlterMetaJob(List<AlterClause> alterClauses, Database db
+                            olapTable.getName(), partitionDuration));
+                    return null;
+                }
+                olapTable.setDataCachePartitionDuration(partitionDuration);
```
1. need write edit log after update.
2. partition duration is also need to update.
`ALTER TABLE table modify partition (p1) set ("xxx"="yyy")`

wanpengfei-git commented:
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

wyb commented on fe/fe-core/src/main/java/com/starrocks/persist/OperationType.java:
```diff
@@ -507,6 +507,9 @@ public class OperationType {
     public static final short OP_ALTER_CATALOG = 13300;
 
+    // alter datacache info of lake table
+    public static final short OP_ALTER_DATACACHE_PARTITION_DURATION = 13400;
```
there is no additional processing for data cache partition duration, so i think there is no need to add a new operation type, and use `OP_ALTER_TABLE_PROPERTIES` is fine.

smartlxh commented on fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:
```diff
@@ -1589,39 +1591,56 @@ public ShowResultSet process(List<AlterClause> alterClauses, Database db, OlapTa
         } else {
             // shouldn't happen
-            throw new DdlException("only support alter enable_persistent_index in shared_data mode");
+            throw new DdlException("alterClause is not a instance of ModifyTablePropertiesClause");
```
The hit of error maybe not clear, for user don't understand what is `alterClause` and `ModifyTablePropertiesClause` which is internal implemention

sduzh commented:
@Mergifyio rebase

sduzh commented:
@Mergifyio rebase

stonechen111 commented on fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java:
```diff
@@ -2993,5 +2993,12 @@ private boolean isEnableFillDataCacheImpl(Partition partition) throws AnalysisEx
         return true;
     }
+
+    public PeriodDuration dataCachePartitionDuration() {
```
OK

stonechen111 commented on fe/fe-core/src/main/java/com/starrocks/catalog/TableProperty.java:
```diff
@@ -776,7 +777,7 @@ public Map<Long, Long> getBinlogAvailaberVersions() {
     }
 
-    public PeriodDuration getDataCachePartitionDuration() {
+    public PeriodDuration dataCachePartitionDuration() {
```
Not necessarily necessary, I will roll it back

stonechen111 committed:
[Feature] Support dynamic modification of datacache.partition_duration

Modify the partition_duration parameter from read-only to dynamic
modification. This facilitates users to modify table attributes in real time.

Signed-off-by: stonechen111 <chenshi@starrocks.com>

sduzh commented:
@Seaven @kangkaisen PTAL
